web/interfaces.js

```diff
@@ -14,6 +14,21 @@
  */
 /* eslint-disable getter-return */
 
+/** @typedef {import("../src/display/api").PDFPageProxy} PDFPageProxy */
+// eslint-disable-next-line max-len
+/** @typedef {import("../src/display/display_utils").PageViewport} PageViewport */
+// eslint-disable-next-line max-len
+/** @typedef {import("./annotation_layer_builder").AnnotationLayerBuilder} AnnotationLayerBuilder */
+// eslint-disable-next-line max-len
+/** @typedef {import("./pdf_rendering_queue").RenderingStates} RenderingStates */
+// eslint-disable-next-line max-len
+/** @typedef {import("./struct_tree_builder").StructTreeLayerBuilder} StructTreeLayerBuilder */
+/** @typedef {import("./text_highlighter").TextHighlighter} TextHighlighter */
+// eslint-disable-next-line max-len
+/** @typedef {import("./text_layer_builder").TextLayerBuilder} TextLayerBuilder */
+/** @typedef {import("./ui_utils").EventBus} EventBus */
+/** @typedef {import("./xfa_layer_builder").XfaLayerBuilder} XfaLayerBuilder */
```

Comment:
This is introducing a *circular dependency* between types, since e.g. `web/annotation_layer_builder.js` references this file, but both `gulp types` and `gulp typestest` works so I assume this is fine!?

Reply:
This is introducing a *circular dependency* between types, since e.g. `web/annotation_layer_builder.js` references types in this file, but both `gulp types` and `gulp typestest` works so I assume this is fine!?